app/controllers/subscriptions_controller.rb

```diff
@@ -5,19 +5,21 @@ class SubscriptionsController < ApplicationController
 
   def create
     @subscription = current_user.subscriptions.build(conference_id: @conference.id)
-    if @subscription.save!
+    save_succeed = @subscription.save
+    if save_succeed
       redirect_to root_path, notice: "You have been subscribed to receive email notifications for #{@conference.short_title}."
     else
-      redirect_to root_path, error: subscription.errors.full_messages.to_sentence
+      redirect_to root_path, error: @subscription.errors.full_messages.to_sentence
     end
   end
 
   def destroy
     @subscription = current_user.subscriptions.find_by(conference_id: @conference.id)
-    if @subscription.destroy
+    if @subscription && @subscription.destroy
       redirect_to root_path, notice: "You have been unsubscribed and now you will not be receiving email notifications for #{@conference.short_title}."
     else
-      redirect_to root_path, error: @subscription.errors.full_messages.to_sentence
+      flash[:error] = 'You are not subscribed to that conference.'
```

Comment:
Yes, the second else is showing an error in case the destroy method fails. 

What you want to do is redirect to root_path with an error unless the subscription exists. 
You can do that in the beginning as you are showing, and then you can leave the destroy if-else as it is.